### PR TITLE
feat: pop-up menu and share page tweaks

### PR DESCRIPTION
**Pull request by jessicaschilling**
## In brief
A grab bag of small-but-helpful changes to IPFS Companion's pop-up menu and share page. This PR saves height of the pop-up overall and gets closer visually to browser-native menu appearance, but also clarifies usage (particularly for onboarding users) through repositioning/rewording items and adding additional info (values for "Copy" menu items!).

### Pop-up header
- Consolidate header area to bring on/off and prefs controls into a more universal (top right) location, and save space
- Make cube/IPFS wordmark combo closer in alignment with the global IPFS logo style overall
- Bring version number next to "IPFS" for clarity, to save space, and to emphasize the gateway/API/peers status info
- List peers first in status items list
- Adds hover titles for all items for increased clarity and beginner-friendliness
- Remove starfield background to make small text easier to read and to bring more into alignment with our gradual move to a "flatter" look

### Pop-up menu
- Removes "Tools" title from tools menu, and moves tools menu above active tab menu 
- Removes global gateway toggle per conversations with @olizilla and @lidel (don't panic, it's still in preferences page)
- Renames remaining tools menu items to "Quick Import/Share..." and "Go to My Node..." for clarity and to remove reference to Web UI 
- Renames active tab menu title to "Current tab" and restyles for consistency with native/browser menu styling (Title Case, gray text)
- Makes menu text smaller throughout for consistency with native/browser menu styling
- Makes menu spacer line lighter gray for consistency with native/browser menu styling
- Makes menu items Title Case for consistency with native/browser menu styling (also cleans up native OS notifications for consistency of Title Case and sentence case)
- Adds option of `helperText` for all menu items, so it can be used for displaying truncated version of items to be copied (shareable link, content path, CID)
- Partially links `helperText` to those items' real values, **but doesn't get all the way** (see "still missing" below)

### Share page
- Give share page title its own i18n message distinct from menu item title, for clarity

## Still missing ...

- Copy helper text isn't dynamic! This info (shareable links, IPFS paths, CIDs to appear under "Copy ..." menu items) comes from copier.js, and I hit a wall with this one. **@lidel or @olizilla -- might you be willing to update the placeholder text in this PR with real live info?**
- Still a few changes to be made to i18n messages, for clarity in both prefs page and menu items around pinning and local gateway language -- but want to make these in a different PR to avoid merge conflict with https://github.com/ipfs-shipyard/ipfs-companion/pull/903 now that's in master

## Screenshots

**"On" state for a DNSLink page**, the most complex display scenario (L-R: Firefox before, Firefox after, Chrome after)
image

**"Off" state** (L-R: Firefox before, Firefox after, Chrome after)
image

**Share page** (L-R: Firefox before, Firefox after)
image

## Issues resolved/partially resolved
This work draws from a variety of longstanding "discussion thread" issues and will close or partially close the following:
- https://github.com/ipfs-shipyard/ipfs-companion/issues/689
- https://github.com/ipfs-shipyard/ipfs-companion/issues/741

### CODE REVIEW

**Comment by jessicaschilling** [comment]
@lidel -- for review ... I hope I'm not trampling on your baby!

**Comment by jessicaschilling** [comment]
(Note: Some of this work will require changing text on the prefs page -- not doing that in this PR to avoid a merge conflict with prior prefs page changes. It's on my list though!)

**Comment by jessicaschilling** [comment]
Hi @lidel -- per our conversation, gently pushing this back over in your direction. Thank you for the review!

File: add-on/_locales/en/messages.json [hunk 1]
```diff
@@ -36,59 +48,47 @@
     "description": "A label tooltip in Node status section of Browser Action pop-up (panel_statusSwarmPeersTitle)"
   },
   "panel_quickImport": {
-    "message": "Share files via IPFS",
+    "message": "Quick Import/Share...",
```
Comment (lidel) [inline add-on/_locales/en/messages.json:51]: Elipsis may look bit better?
```suggestion
    "message": "Quick Import/Share…",
```

File: add-on/_locales/en/messages.json [hunk 2]
```diff
@@ -36,59 +48,47 @@
...
     "description": "A menu item in Browser Action pop-up (panel_quickImport)"
   },
   "panel_openWebui": {
-    "message": "Open Web UI",
+    "message": "Go to My Node...",
```
Comment (lidel) [inline add-on/_locales/en/messages.json:55]:
```suggestion
    "message": "Go to My Node…",
```

File: add-on/src/popup/browser-action/gateway-status.js [hunk 3]
```diff
@@ -27,31 +27,27 @@ module.exports = function gatewayStatus ({
...
-    label: 'panel_statusSwarmPeers',
-    labelLegend: 'panel_statusSwarmPeersTitle',
-    value: swarmPeers,
-    check: swarmPeers,
-    valueClass: 'fw9'
-  })}
+    ${statusEntry({
+      label: 'panel_statusSwarmPeers',
+      labelLegend: 'panel_statusSwarmPeersTitle',
+      value: swarmPeers,
```
Comment (lidel) [inline add-on/src/popup/browser-action/gateway-status.js:33]: Random idea: would it be useful if we make this peer count clickable with `cursor: help` to open `#/peers` in WebUI?

items below could open Preferences and scroll to relevant sections (`/dist/options/options.html#gateway`  and `/dist/options/options.html#api`)

File: add-on/src/popup/browser-action/header.js [hunk 4]
```diff
@@ -1,46 +1,53 @@
...
 const optionsIcon = require('./options-icon')
+const ipfsVersion = require('./ipfs-version')
 const gatewayStatus = require('./gateway-status')
 
 module.exports = function header (props) {
   const { ipfsNodeType, active, onToggleActive, onOpenPrefs, isIpfsOnline, onOpenWelcomePage } = props
   return html`
-    <div class="pt3 pb1 br2 br--top ba bw1 b--white" style="background-image: url('../../../images/stars.png'), linear-gradient(to bottom, #041727 0%,#043b55 100%); background-size: 100%; background-repeat: repeat;">
-      <div class="no-user-select">
+    <div class="br2 br--top ba bw1 b--white" style="background: linear-gradient(to bottom, #041727 0%,#043b55 100%);">
```
Comment (lidel) [inline add-on/src/popup/browser-action/header.js:14]: I think ipfs-css has something like `ipfs-gradient-0` – could we reuse it here?

Comment (jessicaschilling) [reply add-on/src/popup/browser-action/header.js:14]: Ah yes! Gradient direction is reversed but it's still a good luck. Thanks for remembering 😊

File: add-on/_locales/en/messages.json (hunk 2 above)
Comment (jessicaschilling) [reply add-on/_locales/en/messages.json:55]: 🙏

File: add-on/_locales/en/messages.json (hunk 1 above)
Comment (jessicaschilling) [reply add-on/_locales/en/messages.json:51]: 🙏

File: add-on/src/popup/browser-action/gateway-status.js (hunk 3 above)
Comment (jessicaschilling) [reply add-on/src/popup/browser-action/gateway-status.js:33]: Do you mind if we defer on that until we see how the new design is received? Once I got a help cursor on each of those elements, the clickable items were really close to each other spatially -- it's fairly easy to click on the wrong thing.

**Comment by jessicaschilling** [comment]
Hi @lidel --
Re-requesting review after making all but one of the changes you requested (clickable status values, I'm worried they may be more cluttering than helpful).

Do you mind having a more detailed look at this point? I've edited the original comment above with final screenshots and one request for (simple) help with something I'm having trouble with.

Thank you so much!

**Comment by lidel** [comment]
I wired up the hints, they look great! 

> 

## Performance caveat when using without redirect

When local gateway is used, lookups are cheap, because data loaded in the tab is already on local IPFS Node 

CID resolution is much more expensive when redirect is off (eg. when embedded js-ipfs is used). 
In that scenario UI may freeze until IPFS node is able to resolve CID for current URL.

This may take from a few seconds to minutes, and means broken state like this:

> need-async-2020-07-08--14-53-14

I'll look into ways we can do CID resolution asynchronously, so its lack does ot block UI render.

**Comment by jessicaschilling** [comment]
@lidel I made a small change to the disabled-state message, but LGTM! Please feel free to merge if you're happy with it. Thank you!

File: add-on/_locales/en/messages.json [hunk 5]
```diff
@@ -36,61 +48,73 @@
...
+    "message": "Pin this page's IPFS resources to your node to have a local copy that's available offline and never thrown away.",
     "description": "A menu item tooltip in Browser Action pop-up (panel_pinCurrentIpfsAddressTooltip)"
   },
   "panelCopy_currentIpfsAddress": {
-    "message": "Copy IPFS content path",
+    "message": "Copy IPFS Content Path",
     "description": "A menu item in Browser Action pop-up and right-click context menu (panelCopy_currentIpfsAddress)"
   },
+  "panelCopy_currentIpfsAddressTooltip": {
+    "message": "A pathed link that you can use with any gateway",
```
Comment (lidel) [inline add-on/_locales/en/messages.json:95]: @jessicaschilling first time I see "pathed link", and this is more useful in native IPFS tools than gateways, so perhaps we could avoid introducing new terms and go with something like this? (feel free to rephrase) 

```suggestion
    "message": "A canonical content path that you can use with IPFS tools and gateways",
```

Comment (jessicaschilling) [reply add-on/_locales/en/messages.json:95]: That's much nicer 😊

**Comment by jessicaschilling** [comment]
@lidel - Thanks for the resolving the conflicts, and supplying better link-copy text. I think we're good to go here!

**Comment by Mithgol** [comment]
> Removes global gateway toggle per conversations with @olizilla and @lidel (don't panic, it's still in preferences page)

Congratulations, y'all have just made some quick decisions (“Does the global gateway currently have a copy of this IPFS resource? Can I share its global URL in Telegram and expect Telegram to generate a preview under my message? Can I share its global URL in Twitter and expect Twitter to generate a Twitter Card under my message?”) significatly less quicker.

This toggle is not even on the first page of the preferences. It becomes necessary to **scroll** the preferences.

**Comment by lidel** [comment]
@Mithgol I make similar checks by opening IPFS resource in Incognito mode. That way you can check how resource loads from public gateway without the need to disable anything in IPFS Companion.

For a longer rationale, see my comment at https://github.com/ipfs-shipyard/ipfs-companion/issues/910#issuecomment-657841475 – hope this helps.

**Pull request merged**